vital/types/metadata.cxx

```diff
@@ -56,18 +56,17 @@ namespace vital {
       : metadata_item( "Requested metadata item is not in collection", 0, VITAL_META_UNKNOWN )
     { }
 
-    virtual bool is_valid() const { return false; }
-    virtual vital_metadata_tag tag() const { return static_cast< vital_metadata_tag >(0); }
-    virtual std::type_info const& type() const { return typeid( void ); }
-    virtual std::string as_string() const { return "--Unknown metadata item--"; }
-    virtual double as_double() const { return 0; }
-    virtual double as_uint64() const { return 0; }
-    virtual std::ostream& print_value(std::ostream& os) const
+    bool is_valid() const override { return false; }
+    std::type_info const& type() const override { return typeid( void ); }
+    std::string as_string() const override { return "--Unknown metadata item--"; }
+    std::ostream& print_value(std::ostream& os) const override
     {
       os << this->as_string();
       return os;
     }
 
+    metadata_item* clone() const override { return const_cast< unknown_metadata_item* >( this ); }
```

Comment:
Given why this is needed, this seems dangerous. I wonder if it wouldn't be better to have this return `nullptr`? The implication of this method is that the resulting object can be `delete`d, but that is not the case here...